### PR TITLE
Kotlin idioms

### PR DESCRIPTION
**Pull request by Goooler**
Fixes all Detekt issues and replaces some Java APIs with Kotlin idioms.

Follow up #924.

### CODE REVIEW

File: src/main/kotlin/com/github/spotbugs/snom/Confidence.kt [hunk 1]
```diff
@@ -45,24 +42,25 @@ import org.gradle.api.tasks.Internal
...
-        override fun toCommandLineOption(): Optional<String> = Optional.of("-high")
-    }, ;
+        override val commandLineOption: String = "-high"
+    },
+    ;
 
-    @Internal("This is internally used property so no need to refer to judge out-of-date or not.")
-    abstract fun toCommandLineOption(): Optional<String>
+    @get:Internal("This is internally used property so no need to refer to judge out-of-date or not.")
+    internal abstract val commandLineOption: String?
```
Comment (Goooler) [inline src/main/kotlin/com/github/spotbugs/snom/Confidence.kt:65]: We have no need to use `Optional` after migrating to Kotin, right?

File: src/main/kotlin/com/github/spotbugs/snom/internal/SpotBugsRunner.kt [hunk 2]
```diff
@@ -32,115 +32,102 @@ abstract class SpotBugsRunner {
...
-                    "Could not create auxiliary classpath file for SpotBugsTask at " +
-                        auxClasspathFile.normalize().toString(),
-                    e,
-                )
+            auxClasspathFile.parent.createDirectories()
+            if (!auxClasspathFile.exists()) {
+                auxClasspathFile.createFile()
             }
-        } catch (e: Exception) {
-            throw GradleException("Could not create auxiliary classpath file for SpotBugsTask", e)
```
Comment (Goooler) [inline src/main/kotlin/com/github/spotbugs/snom/internal/SpotBugsRunner.kt:143]: There are 2 cathing, but seems they are catching for the same `IOException`, might be a typo here? I've replaced with `catch (e: IOException)`.

File: src/test/kotlin/com/github/spotbugs/snom/SpotBugsPluginSpec.kt [hunk 3]
```diff
@@ -18,7 +18,7 @@ class SpotBugsPluginSpec : DescribeSpec({
         }
 
         it("does not support Gradle 7.0") {
-            shouldThrow<IllegalArgumentException> {
+            shouldThrow<IllegalStateException> {
```
Comment (Goooler) [inline src/test/kotlin/com/github/spotbugs/snom/SpotBugsPluginSpec.kt:21]: Use `check` function the throw `IllegalStateException`.

**Comment by Goooler** [comment]
This depends on #1064, merge that first.

File: src/main/kotlin/com/github/spotbugs/snom/SpotBugsBasePlugin.kt [hunk 4]
```diff
@@ -167,6 +125,7 @@ class SpotBugsBasePlugin : Plugin<Project> {
          * Supported Gradle version described at [official manual site](http://spotbugs.readthedocs.io/en/latest/gradle.html).
          * The convention API provides replacement from 7.1 and later, so we use this value as minimal required version.
          */
+        @Suppress("MaxLineLength")
```
[the remark this answers is not in the document]
Comment (Goooler) [reply src/main/kotlin/com/github/spotbugs/snom/SpotBugsBasePlugin.kt:128]: Nope, this comment is over-length, add this to suppress the check from Detekt.

File: src/main/kotlin/com/github/spotbugs/snom/internal/SpotBugsRunnerForHybrid.kt [hunk 5]
```diff
@@ -114,15 +110,15 @@ class SpotBugsRunnerForHybrid(
...
                     .map(RegularFile::getAsFile)
                     .map(File::toPath)
                     .map(Path::toUri)
                     .map(URI::toString)
-                    .collect(Collectors.toList())
+                    .toList()
                 if (reportPaths.isNotEmpty()) {
                     append(" See the report at: ")
-                    append(reportPaths.joinToString(", "))
+                    append(reportPaths.joinToString())
```
[the remark this answers is not in the document]
Comment (Goooler) [reply src/main/kotlin/com/github/spotbugs/snom/internal/SpotBugsRunnerForHybrid.kt:121]: It's the default param.

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html

File: src/main/kotlin/com/github/spotbugs/snom/internal/SpotBugsRunnerForJavaExec.kt [hunk 6]
```diff
@@ -45,29 +46,22 @@ class SpotBugsRunnerForJavaExec @Inject constructor(
...
-                        }
+                        .map(Provider<File>::get)
                         .map(File::toPath)
                         .map(Path::toUri)
                         .map(URI::toString)
                         .toList()
                     if (reportPaths.isNotEmpty()) {
                         append("See the report at: ")
-                        append(reportPaths.joinToString(", "))
+                        append(reportPaths.joinToString())
```
[the remark this answers is not in the document]
Comment (Goooler) [reply src/main/kotlin/com/github/spotbugs/snom/internal/SpotBugsRunnerForJavaExec.kt:64]: Same above.

File: src/main/kotlin/com/github/spotbugs/snom/SpotBugsBasePlugin.kt [hunk 7]
```diff
@@ -130,35 +104,19 @@ class SpotBugsBasePlugin : Plugin<Project> {
...
     fun verifyGradleVersion(version: GradleVersion) {
-        if (version < SUPPORTED_VERSION) {
-            val message =
-                String.format(
-                    "Gradle version %s is unsupported. Please use %s or later.",
-                    version,
-                    SUPPORTED_VERSION,
-                )
-            throw IllegalArgumentException(message)
+        check(version >= SUPPORTED_VERSION) {
```
[the remark this answers is not in the document]
Comment (Goooler) [reply src/main/kotlin/com/github/spotbugs/snom/SpotBugsBasePlugin.kt]: efceb42552bff8d5a51cfea5c09ba7396085a74f

**Comment by Goooler** [comment]
If we are ready to land this, may merge #1064 first, then rebase this to check the API changes.

File: src/test/kotlin/com/github/spotbugs/snom/SpotBugsPluginSpec.kt (hunk 3 above)
Comment (Goooler) [reply src/test/kotlin/com/github/spotbugs/snom/SpotBugsPluginSpec.kt]: Replaced by efceb42552bff8d5a51cfea5c09ba7396085a74f.

**Review by KengoTODA — APPROVED** [review]
Thanks for your great contribution!

**Pull request merged**